lib/generators/module/module_generator.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'rails/generators'
+
+class ModuleGenerator < Rails::Generators::NamedBase
+  source_root File.expand_path('templates', __dir__)
+
+  def create_app
+    path = "modules/#{file_name}/app"
+    template 'app/controllers/controller.rb.erb',
+             File.join(path, 'controllers', file_name, 'v0', "#{file_name}_controller.rb")
+    template 'app/controllers/application_controller.rb.erb',
+             File.join(path, 'controllers', file_name, 'application_controller.rb')
+    template 'app/models/resource.rb.erb', File.join(path, 'models', file_name, "#{file_name}.rb")
+    template 'app/serializers/serializer.rb.erb',
+             File.join(path, 'serializers', file_name, "#{file_name}_serializer.rb")
+    template 'app/services/configuration.rb.erb', File.join(path, 'services', file_name, 'configuration.rb')
+    template 'app/services/service.rb.erb', File.join(path, 'services', file_name, 'service.rb')
+  end
+
+  def create_lib
+    path = "modules/#{file_name}/lib"
+    template 'lib/namespace/engine.rb.erb', File.join(path, file_name, 'engine.rb')
+    template 'lib/namespace/version.rb.erb', File.join(path, file_name, 'version.rb')
+    template 'lib/tasks/tasks.rake.erb', File.join(path, 'tasks', "#{file_name}_tasks.rake")
+    template 'lib/namespace.rb.erb', File.join(path, "#{file_name}.rb")
+  end
+
+  def create_config
+    path = "modules/#{file_name}"
+    template 'bin/rails.erb', File.join(path, 'bin', 'rails')
+    chmod File.join(path, 'bin', 'rails'), 0o755
+    template 'config/routes.rb.erb', File.join(path, 'config', 'routes.rb')
+    template 'gemspec.erb', File.join(path, "#{file_name}.gemspec")
+    template 'Rakefile.erb', File.join(path, 'Rakefile')
+    template 'Gemfile.erb', File.join(path, 'Gemfile')
```

Comment:
I don't see the `README` anywhere.  Should that be included here (or somewhere else in this file)?